### PR TITLE
Switch to releaser v2

### PR DESCRIPTION
**Pull request by blink1073**
## References
Updates JupyterLab to be compatible with Jupyter Releaser v2

## Code changes
Updates workflows and releaser config.

## User-facing changes
None

## Backwards-incompatible changes
None

### CODE REVIEW

**Comment by jtpio** [comment]
For reference there is a section in the `mdformat` docs about the differences between `mdformat` and `prettier`: https://mdformat.readthedocs.io/en/stable/#why-not-use-prettier-instead

**Comment by blink1073** [comment]
We also needed mdformat because releaser uses it to format the changelog, which caused the pre-commit check to fail when prettier tried to alter it.

**Comment by jtpio** [comment]
This might then also fix https://github.com/jupyterlab/jupyterlab/issues/12671

**Comment by blink1073** [comment]
@meeseeksdev please backport to 3.6.x

File: RELEASE.md [hunk 1]
```diff
@@ -247,50 +247,50 @@ These lines:
 
 ## Post release candidate checklist
 
-- [ ] Modify and run `python scripts/milestone_check.py` to check the issues assigned to this milestone
-- [ ] Write [release highlights](CHANGELOG.md), starting with:
+- \[ \] Modify and run `python scripts/milestone_check.py` to check the issues assigned to this milestone
+- \[ \] Write [release highlights](CHANGELOG.md), starting with:
```
[the remark this answers is not in the document]
Comment (blink1073) [reply RELEASE.md:251]: Thanks!  https://github.com/jupyterlab/jupyterlab/pull/13335

**Pull request merged**